pandas/core/arrays/interval.py

```diff
@@ -1435,6 +1438,40 @@ def contains(self, other):
             other < self._right if self.open_right else other <= self._right
         )
 
+    def isin(self, values) -> np.ndarray:
+        if not hasattr(values, "dtype"):
+            values = np.array(values)
+        values = extract_array(values, extract_numpy=True)
+
+        if is_interval_dtype(values.dtype):
+            if self.closed != values.closed:
+                # not comparable -> no overlap
+                return np.zeros(self.shape, dtype=bool)
+
+            if is_dtype_equal(self.dtype, values.dtype):
+                # error: "ArrayLike" has no attribute "view"  [attr-defined]
+                left = self._combined.view("complex128")  # type:ignore[attr-defined]
+                right = values._combined.view("complex128")
+                return np.in1d(left, right)
+
+            elif needs_i8_conversion(self.left.dtype) ^ needs_i8_conversion(
+                values.left.dtype
+            ):
+                # not comparable -> no overlap
+                return np.zeros(self.shape, dtype=bool)
+
+        return isin(self.astype(object), values.astype(object))
+
+    @property
+    def _combined(self) -> ArrayLike:
+        left = self.left._values.reshape(-1, 1)
+        right = self.right._values.reshape(-1, 1)
+        if needs_i8_conversion(left.dtype):
```

Comment:
why not just *always* do left._concat_same_type to make this simpler? (could have the i8 check inside that if needbe)

Reply:
because we want to end up with shape (N, 2), not (2N,).

(back when IntervalArray was briefly backed by a single array, that array was exactly `_combined`)

Reply:
right but i am saying why don't you just left_concat_same_type then reshape?

Reply:
because then we'd have the wrong contiguity